samples/matter/common/src/ota_multi_image_downloader.h

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#pragma once
+
+#include <zephyr/storage/stream_flash.h>
+
+#include <platform/CHIPDeviceLayer.h>
+
+#include <array>
+
+using namespace chip;
+
+/* A helper class to facilitate the multi variant application OTA DFU
+   (Thread/WiFi switchable)  with the stream_flash backend */
+class OTAMultiImageDownloader {
+public:
+	OTAMultiImageDownloader(int imageId, size_t offset, size_t size, size_t targetOffset, size_t targetSize)
+		: mImgId(imageId), mOffset(offset), mImgSize(size), mTargetOffset(targetOffset),
+		  mTargetImgSize(targetSize)
+	{
+	}
+
+	int Init(const device *flash, ssize_t size);
+	int Write(const uint8_t *chunk, size_t chunkSize);
+	int Finalize();
+	int Apply(const device *flash);
+	int GetImageId() { return mImgId; }
+	void Cleanup() { Platform::MemoryFree(mBuffer); }
+
+private:
+	int Init(const device *flash, size_t offset, size_t size);
+
+	int mImgId;
+	const size_t mOffset;
```

Comment:
All consts should start with k prefix.

Reply:
I thought it was only applicable to *real* constants, like global/static/compile time ones. This one is a class member and the `const` here implies that it's immutable and once initialized in c'tor it should not be changed. So I'm not sure about the coding standard here.

Reply:
I would agree that `k` should be used for `constexpr`-type of constants - known at compile-time :) That's how they are used in the upstream.

Reply:
Ok, so I misinterpreted it and thought we should always use k with consts.